legal-api/src/legal_api/services/search_service.py

```diff
@@ -0,0 +1,216 @@
+            if state in filing_name_to_value and filing_name_to_value[state] in cls.FILINGS_ELIGIBLE_STATES
+        ]
+        return business_states, filing_states
+
+
+    def try_parse_legal_type(legal_type: str) -> bool:
+        try:
+            Business.LegalTypes(legal_type)
+            return True
+        except ValueError:
+            return False
+
+    def separate_legal_types(legal_types: List[str]) -> Tuple[List[str], List[str]]:
+        """
+        Separates input legal types into valid and invalid based on the Business.LegalTypes enum.
+        """
+        input_normalized = [t.strip().upper() for t in legal_types]
+
+        valid_types = [t for t in input_normalized if BusinessSearchService.try_parse_legal_type(t)]
+        invalid_types = [t for t in input_normalized if not BusinessSearchService.try_parse_legal_type(t)]
+        return valid_types, invalid_types
+
+    def paginate(data, page, limit):
+            start = (page - 1) * limit
+            end = start + limit
+            return data[start:end]
+    
+    @staticmethod
+    def get_search_filtered_businesses_results(business_json, identifiers=None, search_filter_name=None, search_filter_type=None,search_filter_status=None):
+        """Return contact point from business json."""
+        
+        filters = []
+
+        if identifiers and isinstance(identifiers, list):
+            filters.append(Business._identifier.in_(identifiers))
+
+        if search_filter_name:
+            filters.append(Business.legal_name.ilike(f'%{search_filter_name}%'))
+
+        if isinstance(search_filter_type, list):
+            valid_filter_values, invalid_filter_values = BusinessSearchService.separate_legal_types(search_filter_type)
+
+            if valid_filter_values:
+                filters.append(Business.legal_type.in_(valid_filter_values))
+    
+        business_states, _ = BusinessSearchService.separate_states_by_type(search_filter_status)
+
+        if search_filter_status and isinstance(search_filter_status, list):
+            filters.append(Business.state.in_(business_states))
+
+        if not filters:
+            return {}
+
+        try:
+            bus_query = db.session.query(Business).filter(*filters)
+            bus_results = []
+
+            for business in bus_query.all():
+                business_json = business.json(slim=True)
+
+                if business.legal_type in (
+                    Business.LegalTypes.SOLE_PROP,
+                    Business.LegalTypes.PARTNERSHIP
+                ):
+                    business_json['alternateNames'] = business.get_alternate_names()
+
+                bus_results.append(business_json)
+
+            return bus_results
+
+        except Exception as e:
+            # Optionally log the error
+            current_app.logger.Error(f'Query error: {e}')
+            return []
+    
+    @staticmethod
+    def get_search_filtered_filings_results(business_json,
```

Comment:
You're fetching and then paginating after? why not just paginate at the DAO level?

Reply:
EG
`q = q.offset((page - 1) * limit).limit(limit)`